libcloud/dns/drivers/digitalocean.py

```diff
@@ -0,0 +1,291 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Digital Ocean DNS Driver
+"""
+
+__all__ = [
+    'DigitalOceanDNSDriver'
+]
+
+from libcloud.utils.py3 import httplib
+
+from libcloud.common.digitalocean import DigitalOcean_v2_BaseDriver
+from libcloud.common.digitalocean import DigitalOcean_v2_Connection
+from libcloud.dns.types import Provider, RecordType
+from libcloud.dns.base import DNSDriver, Zone, Record
+
+
+class DigitalOceanDNSDriver(DigitalOcean_v2_BaseDriver, DNSDriver):
+    connectionCls = DigitalOcean_v2_Connection
+    type = Provider.DIGITAL_OCEAN
+    name = "DigitalOcean"
+    website = 'https://www.digitalocean.com'
+
+    RECORD_TYPE_MAP = {
+        RecordType.NS: 'NS',
+        RecordType.A: 'A',
+        RecordType.AAAA: 'AAAA',
+        RecordType.CNAME: 'CNAME',
+        RecordType.MX: 'MX',
+        RecordType.TXT: 'TXT',
+        RecordType.SRV: 'SRV',
+    }
+
+    def list_zones(self):
+        """
+        Return a list of zones.
+
+        :return: ``list`` of :class:`Zone`
+        """
+        data = self._paginated_request('/v2/domains', 'domains')
+        return list(map(self._to_zone, data))
+
+    def list_records(self, zone):
+        """
+        Return a list of records for the provided zone.
+
+        :param zone: Zone to list records for.
+        :type zone: :class:`Zone`
+
+        :return: ``list`` of :class:`Record`
+        """
+        data = self._paginated_request('/v2/domains/%s/records' % (zone.id),
+                                       'domain_records')
+# TODO: Not use list comprehension to add zone to record for proper data map
+#       functionality?
+        return list(map(self._to_record, data, [zone for z in data]))
```

Comment:
Hm, it looks like `_to_record` takes `zone` as a second argument, but you are passing a list of records as a `zone` argument. 

Or am I missing something?

Also, please use keyword instead of position based arguments where possible to make this more readable and immediately obvious (self._to_record, `data=data, zone=...`).